src/WebJobs.Extensions.Http/Extensions/HttpRequestExtensions.cs

```diff
@@ -63,25 +64,36 @@ public static ClaimsIdentity GetAppServiceIdentity(this HttpRequest request)
                 return null;
             }
             string headerValue = request.Headers[EasyAuthIdentityHeader].First();
-            return FromBase64EncodedJson(headerValue);
+            return FromBase64EncodedJson<ClaimsIdentitySlim>(headerValue, EasyAuthClaimsIdentitySerializer.Value);
         }
 
-        private static ClaimsIdentity FromBase64EncodedJson(string payload)
+        public static ClaimsIdentity GetStaticWebAppsIdentity(this HttpRequest request)
+        {
+            if (!request.Headers.ContainsKey(EasyAuthIdentityHeader))
+            {
+                return null;
+            }
+            string headerValue = request.Headers[EasyAuthIdentityHeader].First();
+            return FromBase64EncodedJson<StaticWebAppsClientPrincipal>(headerValue, StaticWebAppsClaimsIdentitySerializer.Value);
+        }
+
+        private static ClaimsIdentity FromBase64EncodedJson<T>(string payload, DataContractJsonSerializer serializer)
+            where T : IIdentityPrincipal
```

Comment:
The interface isn't technically needed, if this method instead returns the object of type `T` immediately, and then the callers of this method now have a `ClaimsIdentitySlim` or `StaticWebAppsClientPrincipal` that they then just call whatever methods on, instead of needing to code against the interface. I wouldn't be anti-interface necessarily, but I do think it's a bit of unnecessary boilerplate that I don't see us needing the extensibility down the line.